flow/Trace.cpp

```diff
@@ -220,6 +220,28 @@ struct TraceLog {
 		}
 	};
 
+	struct IssuesList : ITraceLogIssuesReporter, ThreadSafeReferenceCounted<IssuesList> {
+		IssuesList(){};
+		void insertIssue(std::string& issue) {
+			MutexHolder h(mutex);
+			issues.insert(issue);
+		}
+
+		std::set<std::string> getAndFlushIssues() {
+			MutexHolder h(mutex);
+			return std::move(issues);
```

Comment:
Is this move safe? I didn't see where we would be creating a new set to replace it, and if we don't then I think it won't be in a well-defined state.

The same question probably applies to other moves as well.

Reply:
Based on my experiments and some discussions I found online(for example [this one]( https://stackoverflow.com/questions/12095048/what-constitutes-a-valid-state-for-a-moved-from-object-in-c11) ), it seems like it's OK to do this. But to be safe I now explicitly reconstruct an empty object on stack after each move. Will update PR once it's correctness clean.